contracts/psp22-traits/lib.rs

```diff
@@ -0,0 +1,53 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+use ink::{
+    env::{DefaultEnvironment, Environment},
+    primitives::AccountId,
+};
+
+pub type Balance = <DefaultEnvironment as Environment>::Balance;
+
+#[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum PSP22Error {
+    InsufficientBalance,
+    InsufficientAllowance,
+}
+
+#[ink::trait_definition]
+pub trait PSP22 {
+    #[ink(message)]
+    fn total_supply(&self) -> Balance;
+
+    #[ink(message)]
+    fn balance_of(&self, owner: AccountId) -> Balance;
+
+    #[ink(message)]
+    fn allowance(&self, owner: AccountId, spender: AccountId) -> Balance;
```

Comment:
aren't we missing `increase_allowance` and `decrease_allowance`? otherwise we have erc20, right?

Reply:
yeah, there's also a byte array in transfer and transfer_from if we want to follow the standard closely. added them